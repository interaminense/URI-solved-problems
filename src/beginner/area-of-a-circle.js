/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 *
 * The formula to calculate the area of a circumference is defined as
 * A = π . R2. Considering to this problem that π = 3.14159:
 *
 * Calculate the area using the formula given in the problem description.
 *
 * INPUT
 *
 * The input contains a value of floating point (double precision), that is the
 * variable R.
 * @param {number} R
 *
 * OUTPUT
 *
 * Present the message "A = " followed by the value of the variable, as in the
 * example bellow, with four places after the decimal point.
 */

const tmpl = result => `A = ${result}`;

const areaOfCircle = (R) => {
  const PI = 3.14159;
  const A = PI * R ** 2;

  return tmpl(A.toFixed(4));
};

export { areaOfCircle, tmpl };
