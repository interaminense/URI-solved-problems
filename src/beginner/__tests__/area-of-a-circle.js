import { areaOfCircle, tmpl } from '../area-of-a-circle';

describe('Area of a circle', () => {
  let R = 0;
  const result = r => areaOfCircle(r);

  it('The raius is 2.0, so, the area of circle should be 12.5664', () => {
    R = 2.0;

    expect(result(R)).toEqual(tmpl((12.5664).toFixed(4)));
    expect(result(R)).toMatchSnapshot();
  });

  it('The raius is 100.64, so, the area of circle should be 31819.3103', () => {
    R = 100.64;

    expect(result(R)).toEqual(tmpl((31819.3103).toFixed(4)));
    expect(result(R)).toMatchSnapshot();
  });

  it('The raius is 150.0, so, the area of circle should be 70685.775', () => {
    R = 150.0;

    expect(result(R)).toEqual(tmpl((70685.775).toFixed(4)));
    expect(result(R)).toMatchSnapshot();
  });
});
