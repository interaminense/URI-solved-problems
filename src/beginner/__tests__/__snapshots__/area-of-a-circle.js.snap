// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area of a circle The raius is 2.0, so, the area of circle should be 12.5664 1`] = `"A = 12.5664"`;

exports[`Area of a circle The raius is 100.64, so, the area of circle should be 31819.3103 1`] = `"A = 31819.3103"`;

exports[`Area of a circle The raius is 150.0, so, the area of circle should be 70685.775 1`] = `"A = 70685.7750"`;
